main.py

```diff
@@ -1,38 +1,69 @@
 from transformers import AutoTokenizer, AutoModelForCausalLM
 import torch
+
 # Convert datasets into a list of strings
 def load_datasets(dataset_path):
     with open(dataset_path) as f:
         text = [l.strip() for l in f if l.strip()]
         return text 
+
 # Set up the device and model
+#Use cuda if available
```

Comment:
generally bad practice to have comments that literally just say what the code does, unless the code is difficult to read. in this case, I dont think the comment adds anything compared to the code `if torch.cuda.is_available()`